torch/testing/_internal/common_methods_invocations.py

```diff
@@ -14758,18 +14758,22 @@ def error_inputs_mean(op_info, device, **kwargs):
                          active_if=(IS_MACOS or IS_WINDOWS)),
         ),
     ),
-    OpInfo(
+    UnaryUfuncInfo(
         'nn.functional.threshold',
-        aten_backward_name='threshold_backward',
-        ref=lambda x, threshold, value: np.where(x > threshold, x, value).astype(x.dtype),
+        ref=lambda x, threshold, value: np.where(np.isnan(x), x, np.where(x > threshold, x, value)).astype(x.dtype),
```

Comment:
why don't you use the same numpy reference as your ref decomposition? 
```suggestion
        ref=lambda x, threshold, value: np.where(x <= threshold, value, x).astype(x.dtype),
```

Reply:
yea, good call. i thought i had tried that without success but it indeed works.